cmake/nuttx/px4_impl_nuttx.cmake

```diff
@@ -297,44 +310,56 @@ function(px4_nuttx_add_export)
+		COMMAND ${CP} -rp ${PX4_SOURCE_DIR}/nuttx-configs/${CONFIG} ${nuttx_build_src}/nuttx/configs
+		COMMAND cd ${nuttx_build_src}/nuttx/tools && sh configure.sh ${CONFIG}/${config_nuttx_config}
+		COMMAND cmake -E touch ${nuttx_configure_stamp}
+		DEPENDS nuttx_patch_${CONFIG} ${nuttx-configs} ${nuttx_copy_stamp}
 		WORKING_DIRECTORY ${PX4_BINARY_DIR}
 		COMMENT "Configuring NuttX for ${CONFIG} with ${config_nuttx_config}")
+	add_custom_target(nuttx_configure_${CONFIG} DEPENDS ${nuttx_configure_stamp} nuttx_patch_${CONFIG})
 
-	# manual reconfigure helpers
+	# manual nuttx oldconfig helper
 	add_custom_target(oldconfig_${CONFIG}
-		COMMAND cd ${nuttx_src}/nuttx
-		COMMAND ${MAKE} -C ${nuttx_src}/nuttx CONFIG_ARCH_BOARD=${CONFIG} oldconfig
-		COMMAND ${CP} ${nuttx_src}/nuttx/.config ${PX4_SOURCE_DIR}/nuttx-configs/${CONFIG}/${config_nuttx_config}/defconfig
+		COMMAND ${MAKE} --no-print-directory -C ${nuttx_build_src}/nuttx CONFIG_ARCH_BOARD=${CONFIG} oldconfig
+		COMMAND ${CP} ${nuttx_build_src}/nuttx/.config ${PX4_SOURCE_DIR}/nuttx-configs/${CONFIG}/${config_nuttx_config}/defconfig
 		COMMAND ${PX4_SOURCE_DIR}/Tools/nuttx_defconf_tool.sh ${PX4_SOURCE_DIR}/nuttx-configs/${CONFIG}/${config_nuttx_config}/defconfig
-		DEPENDS ${nuttx_src}/nuttx/.config
+		DEPENDS nuttx_configure_${CONFIG}
+		WORKING_DIRECTORY ${nuttx_build_src}/nuttx
 		COMMENT "Running NuttX make oldconfig for ${CONFIG} with ${config_nuttx_config}"
 		USES_TERMINAL)
 
+	# manual nuttx menuconfig helper
 	add_custom_target(menuconfig_${CONFIG}
-		COMMAND cd ${nuttx_src}/nuttx
-		COMMAND ${MAKE} -C ${nuttx_src}/nuttx CONFIG_ARCH_BOARD=${CONFIG} menuconfig
-		COMMAND ${CP} ${nuttx_src}/nuttx/.config ${PX4_SOURCE_DIR}/nuttx-configs/${CONFIG}/${config_nuttx_config}/defconfig
+		COMMAND ${MAKE} --no-print-directory -C ${nuttx_build_src}/nuttx CONFIG_ARCH_BOARD=${CONFIG} menuconfig
+		COMMAND ${CP} ${nuttx_build_src}/nuttx/.config ${PX4_SOURCE_DIR}/nuttx-configs/${CONFIG}/${config_nuttx_config}/defconfig
 		COMMAND ${PX4_SOURCE_DIR}/Tools/nuttx_defconf_tool.sh ${PX4_SOURCE_DIR}/nuttx-configs/${CONFIG}/${config_nuttx_config}/defconfig
-		DEPENDS ${nuttx_src}/nuttx/.config
+		DEPENDS nuttx_configure_${CONFIG}
+		WORKING_DIRECTORY ${nuttx_build_src}/nuttx
 		COMMENT "Running NuttX make menuconfig for ${CONFIG} with ${config_nuttx_config}"
 		USES_TERMINAL)
 
-	# build and export
-	add_custom_command(OUTPUT ${nuttx_src}/nuttx/nuttx-export/include/nuttx/config.h
-		COMMAND ${RM} -rf ${nuttx_src}/nuttx/nuttx-export
-		COMMAND ${MAKE} --no-print-directory --quiet -C ${nuttx_src}/nuttx -r CONFIG_ARCH_BOARD=${CONFIG} export > nuttx_build.log
-		DEPENDS ${DEPENDS} ${nuttx_src}/nuttx/.config
+	# nuttx build and export
+	add_custom_command(
+		OUTPUT ${nuttx_export_stamp}
+			${nuttx_export_dir}/include/nuttx/config.h
+		COMMAND ${RM} -rf ${nuttx_export_dir}
+		COMMAND ${MAKE} --no-print-directory --quiet -C ${nuttx_build_src}/nuttx -r CONFIG_ARCH_BOARD=${CONFIG} export ${nuttx_build_output}
```

Comment:
@dagar - thank you, this is almost there. But since the V=1 is not being picked up from the top level.
If you use the same sort of thing you are doing [here](https://github.com/PX4/Firmware/pull/7007/files#diff-23a5ada890263edf0da49d40d9c58f8cR258)
to add ```--no-print-directory --quiet ``` or ```V=1``` we should have a working solution and you can remove the V=1 in the Makefile.